### PR TITLE
Initial implementation of GFT engine.

### PR DESCRIPTION
**Pull request by oliverchang**
Also support fetching multi-target targets (of the format
"binary@target") from FUZZ_TARGET_BUILD_BUCKET_PATH.

### CODE REVIEW

**Comment by oliverchang** [comment]
Friendly ping!

**Comment by oliverchang** [comment]
Friendly ping once more :)

File: src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            'FUZZTEST_REPRODUCERS_OUT_DIR': reproducers_dir,
+        })
+    log_lines = fuzz_result.output.splitlines()
+
+    crashes = []
+    for line in log_lines:
+      reproducer_path = _get_reproducer_path(line)
+      if reproducer_path:
+        crashes.append(
+            engine.Crash(reproducer_path, fuzz_result.output, [],
```
Comment (inferno-chromium) [inline src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py:87]: use named arg, like arg=[]

File: src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                         int(fuzz_result.time_executed)))
+        continue
+
+    # TODO(ochang): Implement this.
```
Comment (inferno-chromium) [inline src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py:91]: nit: can be more specific, like Capture fuzzing statistics ?

Comment (oliverchang) [reply src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py:91]: Done!

File: src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py (hunk 1 above)
Comment (oliverchang) [reply src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py:87]: done!

File: src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+      arguments: Additional arguments needed for reproduction.
+      max_time: Maximum allowed time for the reproduction.
+
+    Returns:
+      A ReproduceResult.
+    """
+    os.chmod(target_path, 0o775)
+    runner = new_process.UnicodeProcessRunner(target_path)
+    result = runner.run_and_wait(
+        timeout=max_time, extra_env={'FUZZTEST_REPLAY': input_path})
```
[the remark this answers is not in the document]
Comment (oliverchang) [reply src/clusterfuzz/_internal/bot/fuzzers/gft/engine.py:111]: I think it'll just be used here, so not much benefit to `FUZZTEST_REPLAY = 'FUZZTEST_REPLAY'`. Can change this later if it's used in more places.

**Pull request merged**